src/main/java/edu/wpi/grip/core/EventHistory.java

```diff
@@ -0,0 +1,43 @@
+package edu.wpi.grip.core;
+
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.events.UndoPublishedEvent;
+import edu.wpi.grip.core.events.UndoableEvent;
+
+import java.util.LinkedList;
+import java.util.Optional;
+import java.util.Queue;
+
+public class EventHistory {
+    private final Queue<UndoableEvent> undoableEvents = new LinkedList<>();
```

Comment:
Why a queue? Wouldn't a stack be better, so we can pop the most recently done action?